main.go

```diff
@@ -395,3 +403,42 @@ func createPullRequestBranch(repo, pr, action string) error {
 	log.Info("Pipeline is expected to start automatically")
 	return nil
 }
+
+func stopStalledPipelines (client *gitlab.Client, vars gitlab.PipelineVariableList) error {
+	integrationPipelinePath := "Northern.tech/Mender/mender-qa"
+
+	/* TODO: check also for gitlab.Pending? */
+	status := gitlab.Running
+	opt := &gitlab.ListProjectPipelinesOptions{
+		Status: &status,
+	}
+
+	pipelines, _, err := client.Pipelines.ListProjectPipelines(integrationPipelinePath, opt, nil)
+	if err != nil {
+		log.Errorf("Could not list running pipelines: %s", err.Error())
+		return err
+	}
+
+	for _, pipeline := range pipelines {
+
+		variables, _, err := client.Pipelines.GetPipelineVariables(integrationPipelinePath, pipeline.ID, nil)
+		if err != nil {
+			log.Errorf("Could not get variables for pipeline: %s", err.Error())
+			return err
+		}
+
+		if reflect.DeepEqual(vars, variables) {
+			log.Infof("Cancelling stalled pipeline %d, url: %s", pipeline.ID, pipeline.WebURL)
```

Comment:
Did you mean "stale", as in "obsolete/superseded"? "Stalled" means "stopped/run to a halt/not progressing".

Reply:
Thank you.